test/e2e/framework/expect.go

```diff
@@ -31,19 +31,37 @@ func ExpectNotEqual(actual interface{}, extra interface{}, explain ...interface{
 }
 
 // ExpectError expects an error happens, otherwise an exception raises
-func ExpectError(err error, explain ...interface{}) {
-	gomega.ExpectWithOffset(1, err).To(gomega.HaveOccurred(), explain...)
+// An additional printf-style explanation must be provided because often errors
+// cannot be understood without it. In cases where the error is known to
+// be descriptive enough in all cases, an empty explainFormatStr can be
+// passed.
+func ExpectError(err error, explainFormatStr string, explainArgs ...interface{}) {
+	if len(explainFormatStr) != 0 {
```

Comment:
nit: I prefer `explainFormatStr != ""` because it is clearer (https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go)

Reply:
All done. Please let me know if there's still something missing or wrong.
Thank you for your patient guidance.